node/node_syncing.go

```diff
@@ -33,6 +38,14 @@ func init() {
 	rand.Seed(time.Now().UnixNano())
 }
 
+// BeaconSyncHook is the hook function called after inserted beacon in downloader
+// TODO: This is a small misc piece of consensus logic. Better put it to consensus module.
```

Comment:
Can you add a TODO to improve the logic?  Leader is way too busy now, may need to choose non-leader to broadcast.

Reply:
Sure, np